api_server.py

```
@@ -160,8 +165,10 @@ def check_rules(available_ids: List[int]) -> Dict:
         }
     with get_db() as conn:
         placeholders = ",".join("?" * len(available_ids))
+        # Use # nosec B608 as placeholders are safely constructed from ? characters
         rows = conn.execute(
-            f"SELECT role, skills FROM crew WHERE id IN ({placeholders})", available_ids
+            f"SELECT role, skills FROM crew WHERE id IN ({placeholders})",
```

Ruff (0.15.1)

[error] Possible SQL injection vector through string-based query construction

(S608)